hers_diagnostic_output/hers_diagnostic_output.py

```diff
@@ -292,213 +1043,228 @@ def multiply_energy_use_and_emission_factors(self):
-            return 0.0
+        if "on_site_power_production" in self.data:
+            return sum(self.data["on_site_power_production"])
+        return 0.0
 
     def calculate_pefrac(self):
         # PEfrac = (TEU - OPP) / TEU
-        TEU = convert(self.calculate_total_energy_use_rated_home(), "kWh", "MBtu")
-        OPP = convert(self.calculate_on_site_power_production(), "kWh", "MBtu")
-        BSL = convert(self.calculate_battery_storage_charge_discharge(), "kWh", "MBtu")
-        return (TEU - OPP + BSL) / TEU
+
+        return (self.teu - self.opp + self.bsl) / self.teu
 
     def calculate_hers_index(self):
         # ERI = PEfrac * (TnML / TRL * IAF_RH) * 100
 
-        PEfrac = self.calculate_pefrac()
-        TnML = self.calculate_total_normalized_modified_load("rated_home")
-        TRL = self.calculate_total_reference_home_load("hers_reference_home")
-        IAF_RH = self.calculate_index_adjustment_factor_rated_home()
-        HERS_INDEX = PEfrac * TnML / (TRL * IAF_RH) * 100
-        return HERS_INDEX
+        return self.pe_frac * self.tnml / (self.trl * self.iaf_rh) * 100
 
     def calculate_carbon_index(self):
         # CO2 Index = ACO2 / ARCO2 * 100
 
-        self.calculate_annual_hourly_co2_emissions()
-        ACO2 = self.emissions["rated_home"]
-        ARCO2 = self.emissions["co2_reference_home"]
-        IAF_RH = self.calculate_index_adjustment_factor_rated_home()
-        return ACO2 / (ARCO2 * IAF_RH) * 100
+        return self.aco2 / (self.arco2 * self.iaf_rh) * 100
 
     def check_index_mismatch(self, index_name, calculated_index, output_index):
         difference_ratio = (calculated_index - output_index) / output_index
         if difference_ratio >= self.INDEX_TOLERANCE:
             raise RuntimeError(
-                f"""\n{self.data["project_name"]} {index_name} outside tolerance.\nCalculated Index: {calculated_index:.2f}\nOutput Index: {output_index:.2f}\nPercent Difference: {difference_ratio*100:.2f}%"""
+                f"""\n{self.project_name} {index_name} outside tolerance.\nCalculated Index: {calculated_index:.2f}\nOutput Index: {output_index:.2f}\nPercent Difference: {difference_ratio*100:.2f}%"""
             )
         else:
-            print(f"""{self.data["project_name"]} {index_name} within tolerance.""")
+            print(f"""{self.project_name} {index_name} within tolerance.""")
 
     def verify_hers_index(self):
         self.check_index_mismatch(
-            "HERS Index", self.calculate_hers_index(), self.data["hers_index"]
+            "HERS Index", self.hers_index, self.data["hers_index"]
         )
 
     def verify_carbon_index(self):
         self.check_index_mismatch(
-            "CO2 Index", self.calculate_carbon_index(), self.data["carbon_index"]
+            "CO2 Index", self.co2_index, self.data["carbon_index"]
         )
 
     def verify(self):
         self.verify_hers_index()
         self.verify_carbon_index()
 
+    def save_hers_index_intermediaries(self):
```

Comment:
I don't think this method belongs in this class. The idea is to make a comparison across multiple HERSDiagnosticData objects. Instead of a method writing this out to a file, you probably just want a function to retrieve the dictionary that you create for the data frame. That way, you can collect them from multiple HERSDiagnosticData objects and create a composite data frame.

Reply:
I changed the function to return a `Dict`.